pkg/api/v1/orchestrator/routes/handlers.go

```diff
@@ -0,0 +1,414 @@
+			Message: "unsupported condition kind: " + string(conditionKind),
+		}
+	}
+
+	conditionID, err := uuid.Parse(c.Param("conditionID"))
+	if err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "invalid conditionID: " + err.Error(),
+		}
+	}
+
+	// the controller retrieved the condition from the queue which created the Task entry
+	// we expect an active condition to allow this publish
+	activeCond, err := r.repository.GetActiveCondition(ctx, serverID)
+	if err != nil {
+		if errors.Is(err, store.ErrConditionNotFound) {
+			return http.StatusBadRequest, &v1types.ServerResponse{
+				Message: err.Error(),
+			}
+		}
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "condition lookup: " + err.Error(),
+		}
+	}
+
+	if !onlyTimestampUpdate && activeCond.ID != task.ID {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: fmt.Sprintf("TaskID: %s does not match active ConditionID: %s", task.ID, activeCond.ID),
+		}
+	}
+
+	// publish Task
+	if err := r.taskKV.publish(
+		ctx,
+		serverID.String(),
+		conditionID.String(),
+		conditionKind,
+		&task,
+		false,
+		onlyTimestampUpdate,
+	); err != nil {
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "Task publish error: " + err.Error(),
+		}
+	}
+
+	return http.StatusOK, &v1types.ServerResponse{
+		Message: "condition Task published",
+	}
+}
+
+// @Summary ConditionPending
+// @Tag Conditions
+// @Description Returns the pending Condition based on the given parameters.
+// @ID conditionPending
+// @Param uuid path string true "Server ID"
+// @Param conditionKind path string true "Condition Kind"
+// @Accept json
+// @Produce json
+// @Success 200 {object} v1types.ServerResponse
+// Failure 400 {object} v1types.ServerResponse
+// Failure 404 {object} v1types.ServerResponse
+// Failure 500 {object} v1types.ServerResponse
+// @Router /servers/{uuid}/condition-pending/{conditionKind} [get]
+func (r *Routes) conditionPending(c *gin.Context) (int, *v1types.ServerResponse) {
+	ctx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.conditionQueuePop")
+	span.SetAttributes(
+		attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(c.Param("uuid"))},
+		attribute.KeyValue{Key: "conditionKind", Value: attribute.StringValue(c.Param("conditionKind"))},
+	)
+	defer span.End()
+
+	serverID, err := uuid.Parse(c.Param("uuid"))
+	if err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "invalid server id: " + err.Error(),
+		}
+	}
+
+	conditionKind := rctypes.Kind(c.Param("conditionKind"))
+	if !r.conditionKindValid(conditionKind) {
+		r.logger.WithFields(logrus.Fields{
+			"kind": conditionKind,
+		}).Info("unsupported condition kind")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "unsupported condition kind: " + string(conditionKind),
+		}
+	}
+
+	conditionRecord, err := r.repository.Get(ctx, serverID)
```

Comment:
why not just call `r.repository.GetActiveCondition` and test that it's pending (like at line 402)?

Reply:
since the match on the `kind` and `Pending` state has to be performed anyway, I went with Get there

Reply:
reworked the handler to use `GetActiveCondition` instead and have the method return the Condition in both `active` or `pending` states.